### DISCUSSION TITLE
Check if any tasks are enqueued when using groups

### DISCUSSION THREAD

**Discussion by philitell** [781flyingdutchman/background_downloader#411] [Q&A]
I'm downloadig a large set of tasks which are divided into different groups (specified in each task). How can i check if all tasks are finished - or at least if any task is enqueued?
As far as i understood the documentation the methods `tasksFinished`, `allTasks` and `allTaskIds` work group-based. So the only way to check if any task is enqueued is to check e.g. `allTasks` for each group. Is that correct?

**Comment by 781flyingdutchman** [comment]
That is correct. In hindsight I think I should have designed the API such that not supplying a `group` parameter to those calls returns all tasks, regardless of group (instead of assuming the `defaultGroup` but that is not what happened, and I can't change it now. If you can think of an elegant way to add this to the API then please let me know - it is trivial to implement on the native side.

**Comment by philitell** [comment]
Thank you for the clarification and thank you for your work on this great package!!

**Comment by 781flyingdutchman** [comment]
Adding `allGroups` parameter to the `allTasks` call that, if set, returns all tasks regardless of group.

**Comment by 781flyingdutchman** [comment]
Added `allGroups` parameter to `allTasks` and `allTaskIds` that returns all tasks regardless of group.

**Comment by philitell** [comment]
Awesome!!!
